wordpress-mobile/WordPress-Android
# Issue/15766 allow reader comment editing

Fixes #15766

This PR allows editing of comments from Reader Comment list for Admin and Editor users of the site that the post belongs to. 

[![Image from Gyazo](https://i.gyazo.com/0a1edc444de2908df8a55335246ab215.gif)](https://gyazo.com/0a1edc444de2908df8a55335246ab215)

Test DB migration script:
- Install app from `trunk` branch.
- Just in case, navigate to reader, load some posts and comments.
- Install this branch on top of `trunk` version.
- Navigate to reader, load some posts and comments.
- Make sure nothing crashes.

To test Editing:
- From Reader, open comment list the post from the site you are Admin or Editor of.
- From the dropdown menu next to the comment, select Edit.
- In editor, edit content of the comment and press Done.
- After you brought back to the comment list, confirm that the changes you made to the comment are reflected in the list.


## Regression Notes
1. Potential unintended areas of impact
- Editing option is available to users with incorrect permission.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing.

3. What automated tests I added (or what prevented me from doing so)
- Piece is legacy code, so pretty difficult to test.

PR submission checklist:
